tests/lac/constraints_make_consistent_in_parallel_01.cc

```diff
@@ -0,0 +1,121 @@
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Test AffineConstraints::make_consistent_in_parallel().
+
+#include <deal.II/base/conditional_ostream.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+
+#include <deal.II/lac/affine_constraints.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+/**
+ * mpirun -np 4 ./constraints_01
+ *
+ * 6-----7-----8
+ * | 0/2 | 1/3 |
+ * 2-----3-----5
+ * | 0/0 | 0/1 |  ... with dof-index, material-id/rank
+ * 0-----1-----4
+ *
+ * Expected output: {3}, {3, 5}, {3, 7}, {3, 5, 7}
+ * Actual output:   {},  {3, 5}, {3, 7}, {3, 5, 7}
+ *
+ * ... code to reproduce https://github.com/dealii/dealii/issues/11725
+ */
+template <typename Number>
+IndexSet
+collect_lines(const AffineConstraints<Number> &constraints,
+              const unsigned int               size)
+{
+  IndexSet lines_local(size);
+  for (const auto &line : constraints.get_lines())
+    lines_local.add_index(line.index);
+  return lines_local;
+}
+
+template <int dim, int spacedim>
+void
+test(const DoFHandler<dim, spacedim> &dof_handler,
+     const IndexSet &                 locally_relevant_dofs)
+{
+  AffineConstraints<double> constraints;
+
+  std::vector<types::global_dof_index> dof_indices(
+    dof_handler.get_fe().n_dofs_per_face());
+
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    for (const auto face : cell->face_indices())
+      if (cell->is_locally_owned() && !cell->at_boundary(face) &&
+          cell->material_id() != cell->neighbor(face)->material_id())
+        {
+          cell->face(face)->get_dof_indices(dof_indices);
+          for (const auto i : dof_indices)
+            constraints.add_line(i);
+        }
+
+  const auto a = collect_lines(constraints, dof_handler.n_dofs());
+  a.print(deallog.get_file_stream());
+
+  constraints.make_consistent_in_parallel(dof_handler.locally_owned_dofs(),
+                                          locally_relevant_dofs,
+                                          dof_handler.get_communicator());
+
+  const auto b = collect_lines(constraints, dof_handler.n_dofs());
+  b.print(deallog.get_file_stream());
+}
+
+int
+main(int argc, char **argv)
+{
+  Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+  MPILogInitAll                    all;
+
+  const int dim = 2;
+
+  ConditionalOStream pcout(std::cout,
+                           Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) ==
+                             0);
+
+  parallel::distributed::Triangulation<dim> tria(MPI_COMM_WORLD);
+  GridGenerator::subdivided_hyper_cube(tria, 2);
+
+  DoFHandler dof_handler(tria);
```

Comment:
Test fails because you didn't specify a template argument. I'll fix that in a follow-up PR.

Reply:
The CI doesn't run all the tests, so it's possible for things like this to get through.